confdb/aclocal_datatype.m4

```diff
@@ -0,0 +1,183 @@
+dnl Utilities in shell functions - reduces size in configure script
+dnl These functions sets $pac_retval
+dnl
+AC_DEFUN([PAC_DATATYPE_UTILS], [
+# Take decimal and turn it into two hex digits
+to_hex() {
+    len=$[]1
+    if test $len -le 9 ; then
+        dnl avoid subshell for speed
+        pac_retval="0${len}"
+    elif test $len -eq 16 ; then
+        pac_retval="10"
+    elif test $len -eq 32 ; then
+        pac_retval="20"
+    else
+        dnl printf is portable enough
+        pac_retval=`printf "%02x" $len`
+    fi
+}
+
+# Convert hex values in the form of 0x######## to decimal for
+# datatype constants in Fortran.
+to_dec() {
+    orig_value=$[]1
+    dnl printf is portable enough
+    pac_retval=`printf "%d" $orig_value`
+}
+
+# return the C type matching integer of size
+get_c_int_type() {
+    len=$[]1
+    pac_retval=
+    for c_type in char short int long "long_long" ; do
+        eval ctypelen=\$"ac_cv_sizeof_$c_type"
+        if test "$len" = "$ctypelen" -a "$ctypelen" -gt 0 ; then
+            if test "$c_type" = "long_long" ; then
+                pac_retval=`echo $c_type | sed -e 's/_/ /g'`
+            else
+                pac_retval=$c_type
+            fi
+            return
+        fi
+    done
+    pac_retval="unavailable"
+}
+
+# return the C type matching integer of size
+get_c_float_type() {
+    len=$[]1
+    pac_retval=
+    for c_type in float double __float128 "long_double" ; do
+        eval ctypelen=\$"ac_cv_sizeof_$c_type"
+        if test "$len" = "$ctypelen" -a "$ctypelen" -gt 0 ; then
+            if test "$c_type" = "long_double" ; then
+                pac_retval=`echo $c_type | sed -e 's/_/ /g'`
+            else
+                pac_retval=$c_type
+            fi
+            return
+        fi
+    done
+    pac_retval="unavailable"
+}
+
+# return (via $pac_retval) the C type matching integer of size
+get_c_bool_type() {
+    len=$[]1
+    pac_retval=
+    for c_type in _BOOL unsigned_char unsigned_short unsigned_int unsigned_long unsigned_long_long ; do
+        eval ctypelen=\$"ac_cv_sizeof_$c_type"
+        if test "$len" = "$ctypelen" -a "$ctypelen" -gt 0 ; then
+            if test "$c_type" != "_BOOL" ; then
+                pac_retval=`echo $c_type | sed -e 's/_/ /g'`
+            else
+                pac_retval=$c_type
+            fi
+            return
+        fi
+    done
+    pac_retval="unavailable"
+}
+])
+
+dnl
+dnl PAC_SET_MPI_TYPE(hexidx, name, len, [multiplier])
+dnl Defines datatype hex handle
+dnl     e.g. PAC_SET_MPI_TYPE(01, MPI_CHAR, 1) -> MPI_CHAR=0x4c000101
+dnl Multiplier can be used to define double types
+dnl     e.g. PAC_SET_MPI_TYPE(48, MPI_COMPLEX, $pac_cv_f77_sizeof_real, 2)
+dnl
+AC_DEFUN([PAC_SET_MPI_TYPE], [
+    if test $3 = 0 ; then
+        $2=MPI_DATATYPE_NULL
+        F77_$2=0
```

Comment:
Currently we set `MPI_DATAYPE_NULL` for unavailable types in `mpif.h`, so this change is unexpected.

Reply:
I misread the original code. Thanks for pointing out. Now fixed.